### PR TITLE
fix(ramp sdk): waiting for ramp network production keys

### LINKED ISSUE #6722: On ramp not working
### What happened?

how dare they! 

image

found by @hassnian 

### Please reproduce in steps

login -> Sidebar -> Add funds

### Expected Behavior

should allow user to buy crypto and FIAT

### What browsers are you seeing the problem on?

Desktop Firefox

### At which address did you encounter bug?

canary.kodadot.xyz, beta.kodadot.xyz

### PR DESCRIPTION
**Pull request by hassnian**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [x] Refactoring

## Context

Just add the correct api key under `RAMP_API_KEY`

- [x] Closes #6722
- [x] Closes https://github.com/kodadot/nft-gallery/issues/6864 

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; 

CleanShot 2023-10-03 at 09 13 54@2x

### <samp>🤖 Generated by Copilot at 01fdac7</samp>

This pull request improves the integration with the Ramp Instant SDK, a service that allows users to buy crypto with fiat. It introduces a new composable `useRamp` that handles the initialization of the SDK with the appropriate options and the API key. It also updates the existing components that use the SDK to leverage the new composable and the runtime config.

### <samp>🤖 Generated by Copilot at 01fdac7</samp>

> _To buy crypto with fiat, we use Ramp_
> _But our code was a bit of a tramp_
> _So we made a composable_
> _Called `useRamp`, so lovable_
> _And refactored our code to be champ_

### CODE REVIEW

File: composables/useRamp.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { Ref, ref } from 'vue'
```
Comment (roiLeo) [inline composables/useRamp.ts:1]:
```suggestion
```
auto imported

File: composables/useRamp.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+interface InitRampParams {
+  address: string
+  defaultAsset: string
+  variant?: AllWidgetVariants
+  showAfterInit?: boolean
+}
+
+export default function useRamp() {
+  const rampInstant: Ref<RampInstantSDK | null> = ref(null)
```
Comment (roiLeo) [inline composables/useRamp.ts:17]:
```suggestion
  const rampInstant = ref<RampInstantSDK | null>(null)
```

File: composables/useRamp.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import { Ref, ref } from 'vue'
+import { RampInstantSDK } from '@ramp-network/ramp-instant-sdk'
+import { AllWidgetVariants } from '@ramp-network/ramp-instant-sdk/dist/types/types'
+import consola from 'consola'
```
Comment (roiLeo) [inline composables/useRamp.ts:4]: `const { $consola } = useNuxtApp()`

**Review by roiLeo — CHANGES_REQUESTED** [review]
Doesn't work for me even with api key 
Screenshot 2023-08-16 at 08-35-22 KodaDot - NFT Market Explorer

**Comment by hassnian** [comment]
@roiLeo  pls review again made the changes that you requested , also did you provide a fresh working api key ?

**Comment by hassnian** [comment]
> > @roiLeo pls review again made the changes that you requested , also did you provide a fresh working api key ?
> 
> I provide the one we were using

I think then we have to wait , a new valid api key is needed https://github.com/kodadot/nft-gallery/pull/6727#issuecomment-1679375853

**Comment by yangwao** [comment]
I've got reply they are already doing check on Kodadot, 🤞 everything will pass soon

**Comment by yangwao** [comment]
image

File: nuxt.config.js [hunk 4]
```diff
@@ -443,6 +443,7 @@ export default defineNuxtConfig({
       baseUrl: process.env.BASE_URL || 'http://localhost:9090',
       googleAnalyticsId: process.env.GOOGLE_ANALYTICS_ID || '',
       dev: process.env.NODE_ENV === 'development',
+      rampApiKey: process.env.RAMP_API_KEY,
```
Comment (yangwao) [inline nuxt.config.js:446]: Seems this one should work, but doesn't after adding demo key 👀

**Comment by yangwao** [comment]
checking your PR, seems it should work, but doesn't suspicious, meanwhile probably will look for other providers probably

https://github.com/kodadot/nft-gallery/pull/6727/files#diff-65d5d69bb59dcc81ad971719549cdee5018f0a240add8b7ef5c65bee3e73dae0R446

**Comment by yangwao** [comment]
Pinged Ramp again, meanwhile we've been approached by another provider, will have updates tomorrow

- #6930

**Comment by yangwao** [comment]
So key is there.. 

image

File: components/common/ConnectWallet/WalletAssetPortfolio.vue [hunk 5]
```diff
@@ -18,21 +18,17 @@
...
-  new RampInstantSDK({
+  initRampInstant({
     defaultAsset: 'KSM',
-    userAddress: accountId.value,
-    hostAppName: 'KodaDot',
-    hostApiKey: 'a99bfvomhhbvzy6thaycxbawz7d3pssuz2a8hsrc', // env
-    hostLogoUrl: 'https://kodadot.xyz/apple-touch-icon.png',
-    variant: 'desktop',
-  }).show()
+    address: accountId.value,
```
Comment (yangwao) [inline components/common/ConnectWallet/WalletAssetPortfolio.vue:30]:
```suggestion
    address: accountId.value,
    url: 'https://app.demo.ramp.network',
```

**Comment by hassnian** [comment]
> I think bc we've got key to this only for now until we will pass KYB 
> 
> - https://docs.ramp.network/testing-environment

afk atm but this won't work url is not a param there gotta pass it here 

image

File: composables/useRamp.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    } = params
+
+    try {
+      rampInstant.value = new RampInstantSDK({
+        defaultAsset,
+        userAddress: address,
+        hostAppName: HOST_APP_NAME,
+        hostApiKey: rampApiKey,
+        hostLogoUrl: HOST_LOGO_URL,
+        variant,
```
Comment (yangwao) [inline composables/useRamp.ts:36]:
```suggestion
        variant,
        url: 'https://app.demo.ramp.network',
```

**Comment by yangwao** [comment]
> afk atm i'd do it myself, but this won't work url is not a param there gotta pass it here

yes, I've checked it's the wrong place 😅 thank you for suggestion!

File: components/common/ConnectWallet/WalletAssetPortfolio.vue [hunk 7]
```diff
@@ -18,21 +18,18 @@
...
+    url: 'https://app.demo.ramp.network',
```
Comment (yangwao) [inline components/common/ConnectWallet/WalletAssetPortfolio.vue:31]:
```suggestion
```

**Comment by yangwao** [comment]
committing from PR like chad, lol. 
Anyway gonna have look on KYBs, both operators are quite slow looks like that

**Comment by yangwao** [comment]
ok, works! 
but before merging we need remove that URL probably.

image

**Review by roiLeo — APPROVED** [review]
✅ loads for me

**Comment by yangwao** [comment]
they are back from "long holiday", so we've implemented feedback they had
- https://github.com/kodadot/nft-gallery/pull/7372

looking forward

**Comment by hassnian** [comment]
@yangwao so now prod keys work ?

**Comment by vikiival** [comment]
Hello can you please pull the latest changes an open pr against `nuxt` branch?

**Comment by hassnian** [comment]
sure

**Comment by hassnian** [comment]
sorry @roiLeo I didn't see you already approved

**Comment by hassnian** [comment]
@yangwao resolved merge conflicts, also removed the   `url: 'https://app.demo.ramp.network',` now it should only need the new api keys

**Comment by yangwao** [comment]
Nah, they are taking too long again to reply with keys. 
I would merge it and then will update keys to production ones on `nuxt` 
I believe we will migrate sooner to `nuxt` than Ramp will get back to me.

Thanks @hassnian !
pay 70 usd

**Pull request merged**